CRM/Core/Payment/PayPalIPN.php

```diff
@@ -321,9 +321,8 @@ public function main() {
       $ids['onbehalf_dupe_alert'] = self::retrieve('onBehalfDupeAlert', 'Integer', 'GET', FALSE);
     }
 
-    $paymentProcessorID = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_PaymentProcessorType',
-      'PayPal_Standard', 'id', 'name'
-    );
+    // CRM-18302: get payment processor ID passed from CRM_Core_Payment_PayPalImpl::doTransferCheckout()
+    $paymentProcessorID = self::retrieve('paymentProcessorID', 'Integer', 'GET', TRUE);
```

Comment:
No need to ensure $paymentProcessorID is set because paymentProcessorID is a required URL parameter and will abort in self::retrieve() if not found, the same way contributionID is required.